### PR TITLE
[RFR][1LP] Test service ansible playbook crud

### PR DESCRIPTION
**Pull request by quarckster**
Intent
=================

New test test_service_ansible_playbook_crud with respectful page model

{{pytest: -v -k "test_service_ansible_playbook_crud" --long-running}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit d7a9aad80e18c8582f479aee26b5e4dd32bc7841:

`cfme/services/catalogs/ansible_catalog_item.py`:
- :red_circle: Line 3:1: **F401** *'widgetastic_patternfly.BootstrapSelect as VanillasBootstrapSelect' imported but unused*
- :red_circle: Line 10:1: **F401** *'utils.appliance.implementations.ui.navigate_to' imported but unused*
- :red_circle: Line 10:1: **F401** *'utils.appliance.implementations.ui.CFMENavigateStep' imported but unused*
- :red_circle: Line 10:1: **F401** *'utils.appliance.implementations.ui.navigator' imported but unused*
- :red_circle: Line 14:23: **F821** *undefined name 'VanillaBootstrapSelect'*
- :red_circle: Line 15:12: **F821** *undefined name 'ParametrizedLocator'*
- :bangbang: Line 54:1: **E302** *expected 2 blank lines, found 1*
- :heavy_exclamation_mark: Line 83:1: **W293** *blank line contains whitespace*

`cfme/tests/services/test_catalog_item.py`:
- :red_circle: Line 147:5: **F841** *local variable 'cat_item' is assigned to but never used*
- :red_circle: Line 152:17: **F821** *undefined name 'catalog'*
- :red_circle: Line 153:16: **F821** *undefined name 'dialog'*
- :heavy_exclamation_mark: Line 154:6: **W292** *no newline at end of file*

Hmmm
:frowning: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit cd7ae55d871468da49a47ccb4ee2869929bfaa72:

`cfme/services/catalogs/ansible_catalog_item.py`:
- :heavy_exclamation_mark: Line 153:1: **W293** *blank line contains whitespace*
- :heavy_exclamation_mark: Line 229:87: **W291** *trailing whitespace*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/services/catalogs/ansible_catalog_item.py [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+class SelectCatalogItemTypeView(ServicesCatalogView):
+    title = Text(".//span[@id='explorer_title_text']")
+    catalog_item_type = BootstrapSelect("st_prov_type", can_hide_on_select=True)
+    add = Button("Add")
+    cancel = Button("Cancel")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_explorer() and
```
Comment (mfalesni) [inline cfme/services/catalogs/ansible_catalog_item.py]: You sure about this one? This usually is a property. And are you sure in_explorer is an appropriate name for this property? Catalog items do not have an "Explorer" :)

File: cfme/services/catalogs/ansible_catalog_item.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+            self.title.text == 'Service Catalog Item "{}"'.format(self.context["object"].name)
+        )
+
+
+class AnsiblePlaybookCatalogItem(Updateable, Navigatable):
+    """Represents Ansible Playbook catalog item.
+
+    Example:
+
+        >>> from cfme.services.catalogs.ansible_catalog_item import AnsiblePlaybookCatalogItem
```
Comment (mfalesni) [inline cfme/services/catalogs/ansible_catalog_item.py]: Use:
```rst
.. code-block:: python

    foo = bar
```

without the leading >>> or ...

File: cfme/services/catalogs/ansible_catalog_item.py [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+        ...         "playbook": "some_playbook.yml",
+        ...         "machine_credential": "CFME Default Credential",
+        ...         "create_new": True,
+        ...         "provisioning_dialog_name": "some_dialog"
+        ...     }
+        ... )
+        >>> catalog_item.create()
+        >>> catalog_item.delete()
+
+    Attributes:
```
Comment (mfalesni) [inline cfme/services/catalogs/ansible_catalog_item.py]: Args

File: cfme/services/catalogs/ansible_catalog_item.py [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+            return True
+
+
+@navigator.register(AnsiblePlaybookCatalogItem, "All")
+class All(CFMENavigateStep):
+    VIEW = AllCatalogItemView
+    prerequisite = NavigateToAttribute("appliance.server", "LoggedIn")
+
+    def step(self):
+        self.prerequisite_view.navigation.select("Services", "Catalogs")
```
Comment (mfalesni) [inline cfme/services/catalogs/ansible_catalog_item.py]: This should be a separate step. I believe such one already exists and you just have to change the prerequisite from LoggedIn to ServicesCatalogs or something like that. Try looking in the cfme.base.ui.

File: cfme/services/catalogs/ansible_catalog_item.py [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+@navigator.register(AnsiblePlaybookCatalogItem, "Details")
+class Details(CFMENavigateStep):
+    VIEW = DetailsAnsibleCatalogItemView
+    prerequisite = NavigateToSibling("All")
+
+    def step(self):
+        tree = self.prerequisite_view.catalog_items.tree
+        tree.click_path(
+            "All Catalog Items",
+            getattr(self.obj.catalog, "name", "Unassigned"),
```
Comment (mfalesni) [inline cfme/services/catalogs/ansible_catalog_item.py]: I believe `.name` should be always defined and then you would rather use: `self.obj.catalog.name or 'Unassigned'`

File: cfme/services/catalogs/ansible_catalog_item.py [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+
+
+@navigator.register(AnsiblePlaybookCatalogItem, "Add")
+class Add(CFMENavigateStep):
+    VIEW = AddAnsibleCatalogItemView
+    prerequisite = NavigateToSibling("All")
+
+    def step(self):
+        self.prerequisite_view.configuration.item_select("Add a New Catalog Item")
+        view = self.prerequisite_view.browser.create_view(SelectCatalogItemTypeView)
```
Comment (mfalesni) [inline cfme/services/catalogs/ansible_catalog_item.py]: This looks like this should be a separate step as well - One step opens the dialog with type selection and next step picks the item.

File: cfme/services/catalogs/ansible_catalog_item.py (hunk 1 above)
Comment (quarckster) [inline cfme/services/catalogs/ansible_catalog_item.py]: Yes, I'm sure, it's not mine code. @sshveta please could you create a jira card to fix this in services code.

File: cfme/services/catalogs/ansible_catalog_item.py (hunk 5 above)
Comment (quarckster) [inline cfme/services/catalogs/ansible_catalog_item.py]: @mfalesni  `self.obj.catalog` can be `None`

Comment (mfalesni) [inline cfme/services/catalogs/ansible_catalog_item.py]: Okay, makes sense then.

**Pull request merged**